scripts/commands/checkLinks.ts

```diff
@@ -94,7 +123,43 @@ async function determineFileBatch(args: Arguments): Promise<FileBatch> {
     toLoad.push("docs/api/qiskit/release-notes/index.md");
   }
 
-  return FileBatch.fromGlobs(toCheck, toLoad);
+  let description: string;
+  if (args.currentApis && args.qiskitReleaseNotes) {
+    description = "non-API docs, current API docs, and Qiskit release notes";
+  } else if (args.currentApis) {
+    description = "non-API docs and current API docs";
+  } else if (args.qiskitReleaseNotes) {
+    description = "non-API docs and Qiskit release notes";
+  } else {
+    description = "non-API docs";
+  }
```

Comment:
We can generate the description without the conditionals using [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) and we can add the historical API docs comment without using an exponential number of conditionals.

```suggestion
  const options: [string, boolean][] = [
    ["non-API docs", true],
    ["current API docs", args.currentApis],
    ["historical API docs", args.historicalApis] , 
    ["Qiskit release notes", args.qiskitReleaseNotes]
  ];

  const optionsEnabled = options.filter(([_, enabled]) => enabled);
  const optionsText = optionsEnabled.map(([text, _]) => text);

  const formatter = new Intl.ListFormat('en', {
    style: 'long',
    type: 'conjunction',
  });
  const description = formatter.format(optionsText);
```

Reply:
Neat! I think this is too cryptic though, so I'm gonna pass on it. But, this does remind me of pattern matching, which I love. If you haven't yet read about pattern matching, check it out - it was added to Python recently: https://peps.python.org/pep-0636/